pkg/addressbook/inmem/inmem.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package inmem
+
+import (
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/addressbook"
+	"github.com/ethersphere/bee/pkg/swarm"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type inmem struct {
+	mtx     sync.Mutex
+	entries map[string]peerEntry // key: overlay in string value, value: peerEntry
+}
+
+type peerEntry struct {
+	overlay   swarm.Address //overlay address
+	multiaddr ma.Multiaddr
+}
+
+func New() addressbook.GetPutter {
+	return &inmem{
+		entries: make(map[string]peerEntry),
+	}
+}
+
+func (i *inmem) Get(overlay swarm.Address) (addr ma.Multiaddr, exists bool) {
+	i.mtx.Lock()
+	defer i.mtx.Unlock()
+
+	val, exists := i.entries[overlay.String()]
```

Comment:
Might be a bit faster to use `string(overlay)` for keys, as there is probably no need for encoding here?

Reply:
not sure by what do you mean with encoding or by faster. in any case, `string(overlay)` does not compile

Reply:
Sorry, `string(overlay.Bytes())`. Because the `String()` method of `swarm.Address` does the encoding, which is not overly expensive, but still is probably a bit more then just casting byte array to string.

Reply:
i prefer to keep use of `String()` in order to preserve the properties that `Address` type provides (that addresses are hex encoded). the solution you're suggesting does not render in the terminal as standard characters and will spit out weird terminal output when a test fails on a value mismatch

Reply:
I do not think that this addressbook implementation should be optimised at this level, as a persistent implementation should be used in production, and this one eventually as a mock. 

But, since we already have similar situation is peerRegistry, would it be useful to have `Address.ByteString` method and `NewFromByteString` constructor for places where performance matters?

Reply:
i think that in this case this is a clear premature optimization :)

Reply:
It is evil! 😂